### ISSUE TITLE
bun test afterEach does not run after each test as supposed to

### ISSUE THREAD

**Issue by simylein** [oven-sh/bun#2487]
### What version of Bun is running?

0.5.8

### What platform is your computer?

Darwin 22.3.0 arm64 arm

### What steps can reproduce the bug?

The following test code should fire logs before and after each test. 
```
import { afterEach, beforeEach, describe, it } from 'bun:test';

beforeEach(() => {
	console.log('beforeEach has fired');
});

afterEach(() => {
	console.log('afterEach has fired');
});

describe('test closure', () => {
	it('some test', () => {
		// nice test here
	});

	it('another test', () => {
		// beautiful test here
	});

	it('more testing', () => {
		// lovely test here
	});

	it('too much testing', () => {
		// marvelous test here
	});
});
```

### What is the expected behavior?

```
bun test v0.5.8 (1a25af5e)
[0.03ms] ".env"

main.spec.ts:
beforeEach has fired
✓ test closure > some test
afterEach has fired
beforeEach has fired
✓ test closure > another test
afterEach has fired
beforeEach has fired
✓ test closure > more testing
afterEach has fired
beforeEach has fired
✓ test closure > too much testing
afterEach has fired

 4 pass
 0 fail
Ran 4 tests across 1 files [9.00ms]
```

### What do you see instead?

```
bun test v0.5.8 (1a25af5e)
[0.03ms] ".env"

main.spec.ts:
beforeEach has fired
✓ test closure > some test
beforeEach has fired
✓ test closure > another test
beforeEach has fired
✓ test closure > more testing
beforeEach has fired
✓ test closure > too much testing

 4 pass
 0 fail
Ran 4 tests across 1 files [9.00ms]
```

### Additional information

I was not sure if the order would be
```
beforeEach has fired
✓ test closure > some test
afterEach has fired
```
or rather
```
beforeEach has fired
afterEach has fired
✓ test closure > some test
```
as long as it is executed. 

Side note: I am really loving this new runtime and its speed 🙂. 
Thanks to all the contributors who make this possible 🙂.

**Comment by simylein** [comment]
I would like to work on this and create a pr. I would need some advice on where to start.

**Comment by simylein** [comment]
This issue seems to be fixed in the latest version 0.6.1. I am therefore closing this issue now. Thank you to whoever fixed this 🙂
